dali/operators/reader/loader/nemo_asr_loader.cc

```diff
@@ -0,0 +1,192 @@
+// Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <string>
+#include "dali/core/common.h"
+#include "dali/core/error_handling.h"
+#include "dali/operators/decoder/audio/generic_decoder.h"
+#include "dali/operators/decoder/audio/audio_decoder_impl.h"
+#include "dali/operators/reader/loader/nemo_asr_loader.h"
+#include "dali/operators/reader/loader/utils.h"
+#include "dali/pipeline/data/views.h"
+#include "dali/core/tensor_shape_print.h"
+#include "dali/pipeline/util/lookahead_parser.h"
+#include "dali/util/file.h"
+
+namespace dali {
+
+namespace detail {
+
+std::string trim(const std::string& str,
+                 const std::string& whitespace = " \t") {
+  const auto str_begin = str.find_first_not_of(whitespace);
+  if (str_begin == std::string::npos)
+    return {};  // no content
+  const auto str_end = str.find_last_not_of(whitespace);
+  const auto str_len = str_end - str_begin + 1;
+  return str.substr(str_begin, str_len);
+}
+
+std::string NormalizeText(std::string& text) {
+  // Remove trailing and leading whitespace
+  auto norm_text = trim(text);
+
+  // Convert to lowercase
+  for (auto &c : norm_text) {
+    if (std::isupper(c)) {
+      c = std::tolower(c);
```

Comment:
I think you can always call `tolower` and skip that if.

Reply:
will do